Add getting started tutorials step 1  through 17 (#257)
MicroEJ/docs: Tutorials/gettingstarted/Step06.rst

Event Handling
==============

> 

Reviewer: Is there a place where you introduce what are events?
Like in the training a very short introduction could help understanding the principles:
Example: 
```suggestion
- MicroUI generates integer-based events that encode the low-level input type and action. The application can handle these events in the ``handleEvent`` method.

```

The handleEvent method
---------------------------

> - Every class that extends Widget inherits the handleEvent method

Reviewer: ```suggestion
- Every class that extends ``Widget`` inherits the ``handleEvent()`` method.

```

Wraps every inlined code, variables, types, methods with ``

> - It is possible to handle events overriding this class on a created sample

Reviewer: ```suggestion
- Add custom event handling by overriding the ``handleEvent()`` method of a widget.

```

> - Using as a sample the Button class

Reviewer: ```suggestion
- Using the Button class from the ``Widget`` library as a sample:

```

.. code:: java

	@Override
	public boolean handleEvent(int event) {
		int type = Event.getType(event);
		if (type == Pointer.EVENT_TYPE) {
			int action = Pointer.getAction(event);
			if (action == Pointer.PRESSED) {
				setPressed(true);
			} else if (action == Pointer.RELEASED) {
				setPressed(false);
				handleClick();
				return true;
			}
		} else if (type == DesktopEventGenerator.EVENT_TYPE) {
			int action = DesktopEventGenerator.getAction(event);
			if (action == PointerEventDispatcher.EXITED) {
				setPressed(false);
			}
		}

		return super.handleEvent(event);
	}

> 

Reviewer: Could you highlight what this code actually do? Either with comments in the code or with a small text below the sample.
I guess this is the first time the reader gets in touch with MicroUI events in the tutorial, so we may explain that this code handle Pointer events PRESSED, RELEASED and manage to call the handleClick() method when a click is detected.

Reviewer: + indicate that we generally end calling super.handleEvent if the event was not handled

> - It's important to know that the widget should be setEnabled(true) in order to events to work

Reviewer: ```suggestion
- It's important to note that only widgets that are "enabled" will receive input events. One can enable a widget by calling ``setEnabled(true)``.

```

> - The handleClick method is where the clickListener is used

Reviewer: ```suggestion
- In the ``Button`` class, the click triggers an action which is defined by the registered ``OnClickListener``. The ``handleClick`` method is where the listener is called:

```

.. code:: java
 
    public void handleClick() {
        OnClickListener listener = this.onClickListener;
        if (listener != null) {
            listener.onClick();
        }
    }

Using events with buttons

> -------------------------------

Reviewer: Same length than title

```suggestion
-------------------------
```

> Events are handled in the same way

Reviewer: I don't get the transition with previous section. Do you want to show the button in action (with the OnClickListener)? If so, please rephrase.

Author: Yes, the goal is to show the Button in action

> - Using the code created in the Last Step, it is possible to simply add the onClick Listener

Reviewer: ```suggestion
- Using the code created in the previous step, let's add a simple action to the button by adding a ``OnClickListener``.
```

.. code:: java

    button.setOnClickListener(new OnClickListener() {
        @Override
        public void onClick() {
            System.out.println("Clicked!");
        }
    });

> This should be shown in the console

Reviewer: ```suggestion
When running the modified sample, this should be shown in the console:
```
